main/src/main/scala/sbt/Main.scala

```diff
@@ -64,6 +64,11 @@ private[sbt] object xMain {
           Seq(defaults, early),
           runEarly(DefaultsCommand) :: runEarly(InitCommand) :: BootCommand :: Nil
         )
+        if (xMain.getClass.getClassLoader.toString == "SbtMetaBuildClassLoader") {
+          val msg = s"sbt was launched with an old version of sbt-launcher. Please upgrade the " +
```

Comment:
Normally sbt is backward and forward compatible against the launcher, so I think there should be a really good reason to print warnings here. I am kind of ok with ppl upgrading to new launcher or script only if they run into an issue.

Reply:
I would normally agree with that, except that failure in something like `consoleProject` is extremely subtle. It was almost pure luck that I was able to debug that when it came up in the gitter channel.

Reply:
Currently the release note says you can upgrade by changing `build.properties` file - https://www.lightbend.com/blog/sbt-1.3.0-release
If we're requiring the script and the launcher to be upgraded, then we need to probably start from there, and also reword that this is a special occurrence. There's been ideas about letting ppl write JVM props in `build.properties`, so we might want to wait for that to come out. (1.4.0, maybe?)

Reply:
The release notes also says right underneath:
> To get other benefits to launching script, please upgrade the sbt script following the setup instruction.

I think we could just strengthen that wording. The truth is that the old launcher does work for probably 99% of use cases, it's just very subtle to debug when it doesn't work.

Reply:
Maybe we can say we _strongly recommend_ upgrading for feature releases (sbt 1.2.x -> 1.3.0).

Reply:
Do you mean in the warning that's printed by sbt in this PR or in the release notes?

Reply:
I meant for the release note, which I did - https://www.lightbend.com/blog/sbt-1.3.0-release

> To get other benefits to launching script and sbt launcher, we strongly recommend upgrading the sbt script following the setup instruction.